### Q1). Print Your Name with your Father name and Date of birth using suitable escape sequence charactor

In [13]:
name = input("My Name is: ")
f_name  = input("My father name is: ")
birht_date =  input("My age is : ")
my_data = name + " " + f_name + " " + birht_date
print(my_data) # by using input value.....

My Name is:  Ahmed Raza
My father name is:  Muhammad Raza
My age is :  26


Ahmed Raza Muhammad Raza 26


In [15]:
My_Data = ["Ahmed Raza", "Muhammad Raza", "25-may-1999"]
print(My_Data) #Creat a list.... 

['Ahmed Raza', 'Muhammad Raza', '25-may-1999']


### Q2). Write your small bio using variables and print it using print function

In [18]:
full_name = "Ahmed Raza"
father = "Muhammad Raza"
age = 26
Hiegth = 5.11
Grade = "Intermediate"
print("----Bio----")
print("Name:",full_name)
print("Father_Name:",father)
print("age:",age)
print("Hieght:",Hiegth)
print("Qualigfication:", Grade)


----Bio----
Name: Ahmed Raza
Father_Name: Muhammad Raza
age: 26
Hieght: 5.11
Qualigfication: Intermediate


### Q3). Write a program in which use all the operators we can use in Python

In [20]:
# Demonstrating All Operators in Python

# 1. Arithmetic Operators
a = 15
b = 4
print("Arithmetic Operators:")
print("a + b =", a + b)   # Addition
print("a - b =", a - b)   # Subtraction
print("a * b =", a * b)   # Multiplication
print("a / b =", a / b)   # Division (float)
print("a // b =", a // b) # Floor Division
print("a % b =", a % b)   # Modulus
print("a ** b =", a ** b) # Exponentiation
print()


# 2. Comparison Operators
print("Comparison Operators:")
print("a == b:", a == b)
print("a != b:", a != b)
print("a > b:", a > b)
print("a < b:", a < b)
print("a >= b:", a >= b)
print("a <= b:", a <= b)
print()

# 4. Logical Operators
p = True
q = False
print("Logical Operators:")
print("p and q:", p and q)
print("p or q:", p or q)
print("not p:", not p)
print()

# 6. Identity Operators
print("Identity Operators:")
list1 = [1, 2, 3]
list2 = [1, 2, 3]
list3 = list1
print("list1 is list3:", list1 is list3)
print("list1 is not list2:", list1 is not list2)
print()

Arithmetic Operators:
a + b = 19
a - b = 11
a * b = 60
a / b = 3.75
a // b = 3
a % b = 3
a ** b = 50625

Comparison Operators:
a == b: False
a != b: True
a > b: True
a < b: False
a >= b: True
a <= b: False

Logical Operators:
p and q: False
p or q: True
not p: False

Identity Operators:
list1 is list3: True
list1 is not list2: True



### Q4). Completes the following steps of small task:
-Mention Marks of English , Islamiat and Maths out of 100 in 3 different variables
-Mention Variable of Total Marks and assign 300 to it
-Calculate Percentage

In [29]:
Eng = 60
math = 80
Isl = 40
obt_marks = Eng + math + Isl
percent = float(obt_marks * 100 / 300)
print("Obtain Marks:", obt_marks)
print("Total Marks: 300")
print("percent:", percent,"%")

Obtain Marks: 180
Total Marks: 300
percent: 60.0 %


### Q5).A company decided to give bonus of 5% to employee if his/her year of service is more than 5 years. Ask user for their salary and year of service and print the net bonus amount.

In [35]:
User_salary = float(input("Your Salary is: "))
User_service = int(input("Your duration: "))
if (User_service >= 5):
    bonus = User_salary * 0.05
    print("you are eligible for bonus of:",bonus)
    print("This Month You Got:", User_salary+bonus)
else:
    print("Sorry You are not eligible for bonus")
    print("Your Current Salary is:",User_salary)

Your Salary is:  60000
Your duration:  2


Sorry You are not eligible for bonus
Your Current Salary is: 60000.0


### Q6). Write a program to check whether a person is eligible for voting or not. (accept age from user) if age is greater than 17 eligible otherwise not eligible

In [32]:
age = int(input("enter your age:"))
if (age >=18 ):
    print("you can vote")
else:
    print("You are not eligible")

enter your age: 20


you can vote


### Q7). Write a program to check whether a number entered by user is even or odd.

In [37]:
num = int(input("Enter any num: "))
if (num % 2 != 0):
    print("It's Odd num:", num)
else:
    print("It's Even num:", num)

Enter any num:  13


It's Odd num: 13


### Q8). Write a program to check whether a number is divisible by 7 or not. Show Answer

In [39]:
num = int(input("Enter a number: "))
if num % 7 == 0:
    print(num, "is divisible by 7.")
else:
    print(num, "is not divisible by 7.")

Enter a number:  49


49 is divisible by 7.


### Q9).Write a program to display "Hello" if a number entered by user is a multiple of five , otherwise print "Bye".

In [42]:
sub = int(input("Enter any num: "))
if (sub >=5):
    print("Hello")
else:
    print("Bye")

Enter any num:  150


Hello


### Q10). Write a program to display the last digit of a number.

In [46]:
num = int(input("Enter a number: "))
last_digit = num % 10
print("The last digit of", num, "is", last_digit)

Enter a number:  675


The last digit of 675 is 5


### Q11). Take values of length and breadth of a rectangle from user and print if it is square or rectangle.

In [47]:
length = float(input("Enter len value: "))
breadth = float(input("Enter breadth value:"))
if (length == breadth):
    print("It is Square")
else:
    print("It is Rectangle")

Enter len value:  45
Enter breadth value: 34


It is Rectangle


### Q12). Take two int values from user and print greatest among them.

In [50]:
val1 = int(input("Enter First value: "))
val2 = int(input("Enter Second value: "))
if (val1 > val2):
    print(val1, "is greater than", val2)
else:
    print(val2, "is greater than", val1)

Enter First value:  43
Enter Second value:  45


45 is greater than 43


### Q13).A shop will give discount of 10% if the cost of purchased quantity is more than 1000. Ask user for quantity Suppose, one unit will cost 100. Judge and print total cost for user.

In [58]:
Unit_Qty = int(input("Your Units Quatity: "))
unit_cost = 100
total_cost = Unit_Qty * 100
print("Your Current amount is: ", total_cost)
if ( Unit_Qty > 1000):
    discount = total_cost * 0.10
    total_cost -= discount
    print("you got 10% discount:",total_cost)
else:
    print("sorry you don't have any discount")

Your Units Quatity:  8000


Your Current amount is:  800000
you got 10% discount: 720000.0


### Q14). A school has following rules for grading system:
a. Below 25 - F

b. 25 to 45 - E

c. 45 to 50 - D

d. 50 to 60 - C

e. 60 to 80 - B

f. Above 80 - A

Ask user to enter marks and print the corresponding grade

In [1]:
marks = float(input("Enter Your Marks: "))
if( marks >= 80 and marks <100):
    print("your Grade is A")
elif( marks >=60 and marks <80):
    print("your Grade is B")
elif( marks >=50 and marks <60):
    print("your Grade is C")
elif( marks >=45 and marks <50):
    print("your Grade is D")
elif( marks >=25 and marks <45):
    print("your Grade is E")
elif( marks <25):
    print("your Grade is Fail")
else:
    print("You Entered Invalid marks")
    

Enter Your Marks:  78


your Grade is B


### Q15).A student will not be allowed to sit in exam if his/her attendence is less than 75%.

Take following input from user

Number of classes held

Number of classes attended.

And print

percentage of class attended

Is student is allowed to sit in exam or not.

In [4]:
class_held = int(input("Enter Total lecture in class: "))
Student_Attendence = int(input("Enter Total Attendence in class: "))
attend_per = int(Student_Attendence*100/class_held)
print(attend_per,"%")
if (attend_per >= 75):
    print("You are allowed in exam")
else:
    print("You are not allowed in exam")

Enter Total lecture in class:  60
Enter Total Attendence in class:  50


83 %
You are allowed in exam


#### Q15p1). Modify the above question to allow student to sit if he/she has medical cause. Ask user if he/she has medical cause or not ( 'Y' or 'N' ) and print accordingly

In [16]:
Total_lecture = int(input("Enter total lecture in semister: "))
Total_attendence = int(input("Enter your total attendence in semister: "))
attendence_per = int(Total_attendence *100/Total_lecture)
print(attendence_per,"%")
if attendence_per >= 75:
    print("You are allow to sit in exam")
else:
    medical_issue = input("is your leave cause madical? (Y/N)").strip().upper()
    if medical_issue == "Y":
        print("You are allow due to medical issue")
    else:
        print("You are not allow to sit in exam")

Enter total lecture in semister:  70
Enter your total attendence in semister:  48


68 %


is your leave cause madical? (Y/N) y


You are allow due to medical issue


### Q16). Write a program to check if a year is leap year or not.
If a year is divisible by 4 then it is leap year but if the year is century year like 2000, 1900, 2100 then it must be divisible by 400.

In [18]:
year = int(input("Enter a year: "))
if (year % 400 == 0) or (year % 4 == 0 and year % 100 != 0):
    print(year, "is a Leap Year.")
else:
    print(year, "is NOT a Leap Year.")


Enter a year:  2017


2017 is NOT a Leap Year.


### Q13). Ask user to enter age, gender ( M or F ), marital status ( Y or N ) and then using following rules print their place of service.
if employee is female, then she will work only in urban areas.

if employee is a male and age is in between 20 to 40 then he may work in anywhere

if employee is male and age is in between 40 t0 60 then he will work in urban areas only.

And any other input of age should print "ERROR"

In [34]:
employee = input("Enter your name: ")
age = int(input("Enter your age: "))
gender = input("what is your gender? answer in (M/F): ").strip().upper()
marital_status = input("Are You married? Answer in (Y/N): ").strip().upper()
if(gender == "F"):
    print("You will work only in urban areas")
elif(gender == "M" and 20 <= age < 40):
    print("You can work anywhere")
elif(gender == "M" and 40 <= age < 60):
    print("You can work only in urban areas")
else:
    print("Error")
    

Enter your name:  Ali
Enter your age:  44
what is your gender? answer in (M/F):  M
Are You married? Answer in (Y/N):  y


You can work only in urban areas


### Q14).Write a program to calculate the electricity bill (accept number of unit from user) according to the following criteria : Unit Price
uptp 100 units no charge Next 200 units Rs 5 per unit After 200 units Rs 10 per unit (For example if input unit is 350 than total bill amount is Rs.3500 (For example if input unit is 97 than total bill amount is Rs.0 (For example if input unit is 150 than total bill amount is Rs.750
Take input of age of 3 people by user and determine oldest and youngest among them.

In [40]:
unit = int(input("Enter total unit in a month: "))
if unit <= 100:
    print("Your Bill amount is : 0")
elif unit >100 and unit <200:
    print("Your bill amount is: ", unit * 5)
elif unit >=200:
    print("Your bill amount is: ", unit * 10)
else:
    print("invalid unit")

Enter total unit in a month:  101


Your bill amount is:  505


In [ ]:
age1 = input("Enter your age: ")
age2 = input("Enter your age: ")
age3 = input("Enter your age: ")
oldest = max(age1, age2, age3)
youngest = min(age1, age2, age3)
print("Oldest age is:",oldest)
print("Youngest age is:",youngest)

ok